app/queue/notifications.py

```diff
@@ -152,6 +178,39 @@ def system_deleted_notification(notification_type, host):
     return result
 
 
+def system_stale_notification(notification_type, host):
+    base_notification_obj = build_base_notification_obj(notification_type, host)
+
+    canonical_facts = host.get("canonical_facts")
+    system_profile = host.get("system_profile_facts")
+    notification = {
+        "context": {
+            "inventory_id": host.get("id"),
+            "hostname": canonical_facts.get("fqdn", ""),
+            "display_name": host.get("display_name"),
+            "rhel_version": build_rhel_version_str(system_profile),
+            # get host url
+            "host_url": "",
+            "tags": host.get("tags"),
+        },
+        "events": [
+            {
+                "metadata": {},
+                "payload": {
+                    "insights_id": canonical_facts.get("insights_id", ""),
+                    "subscription_manager_id": canonical_facts.get("subscription_manager_id", ""),
+                    "satellite_id": canonical_facts.get("satellite_id", ""),
+                    "groups": [{"id": group.get("id"), "name": group.get("name")} for group in host.get("groups")],
+                },
+            },
+        ],
+    }
+
+    notification.update(base_notification_obj)
+    result = SystemStaleSchema().dumps(notification)
+    return result
```

Comment:
tiny optional suggestion, since `result` is only used in this one location
```suggestion
    return SystemStaleSchema().dumps(notification)
```